nixos/modules/system/boot/systemd.nix

```diff
@@ -158,6 +171,7 @@ let
     ] ++ optionals cfg.package.withHostnamed [
       "dbus-org.freedesktop.hostname1.service"
       "systemd-hostnamed.service"
+      "systemd-hostnamed.socket"
```

Comment:
From the release notes:

> systemd-hostnamed now exposes the machine ID and boot ID via D-Bus. It also exposes the hosts AF_VSOCK CID, if available.
>
> systemd-hostnamed now provides a basic Varlink interface.
>
> systemd-hostnamed exports the full data in os-release(5) and machine-info(5) via D-Bus and Varlink.
> 
> hostnamectl now shows the system's product UUID and hardware serial number if known.

Source is https://raw.githubusercontent.com/systemd/systemd/main/units/systemd-hostnamed.socket